cmd/uzfs_test/zrepl_utest.c

```diff
@@ -687,80 +964,256 @@ zrepl_rebuild_test(void *arg)
 		goto exit;
 	}
 
+	/* Copy downgraded replica (ds1) name in mgmt_ack */
 	strncpy(mgmt_ack->dw_volname, ds1, sizeof (mgmt_ack->dw_volname));
-	printf("Healthy replica: %s\n", mgmt_ack->volname);
+
+	printf("Replica being rebuild is: %s\n", mgmt_ack->dw_volname);
+	printf("Replica helping rebuild is: %s\n", mgmt_ack->volname);
 	printf("Rebuilding IP address: %s\n", mgmt_ack->ip);
 	printf("Rebuilding Port: %d\n", mgmt_ack->port);
-	printf("Downgraded replica: %s\n", mgmt_ack->dw_volname);
+
+	/*
+	 * Start rebuild process on downgraded replica ds1
+	 * by sharing IP and rebuild_Port info with ds1.
+	 */
+	rc = zrepl_utest_replica_rebuild_start(new_fd, mgmt_ack,
```

Comment:
new_fd is a connection formed with 'ds0'.. here, we are sending 'rebuild start' for a different dataset on same connection.. this works in this case. But, lets have a new connection and do handshake.

Reply:
new_fd is basically created when replica try to connect to iSCSI controller. Ideally this should be one per IP address, (As per discussion with Jan) but if we think that this is the way to go ahead for UT then I have to find out a smart way to do this. Please do let me know if you think otherwise.